### ISSUE TITLE
TextColumnHandler's length_features treats missing text as the string 'nan' instead of missing

### ISSUE THREAD

**Issue by bijay-odyssey** [bijay-odyssey/edaprep#45]
`TextColumnHandler(strategy="length_features")` converts the raw column with
`X[column].astype(str)` before computing `__length`/`__n_words`, without dropping or
masking missing values first. Pandas' `.astype(str)` turns a `NaN` cell into the
3-character string `"nan"`, so a missing text value gets a real (wrong) length/word
count instead of a missing one.

This is inconsistent within the same file: `text_statistics()` (used for the fit-time
report) correctly does `series.dropna()` before measuring, so the *reported*
statistics and the *actual transform output* disagree about what a missing text value
looks like.

### Reproduced

```python
df = pd.DataFrame({"t": ["hello world", "a longer sentence here", None, "short", None]})
TextColumnHandler(["t"]).fit_transform(df, None, ctx)[["t__length", "t__n_words"]]
# t__length: [11, 22, 4, 5, 4]     <- rows 2 and 4 are None, reported length 4
# t__n_words: [2, 4, 1, 1, 1]      <- reported as 1 word ("nan"), not missing
```

Nothing is counted or reported for this substitution -- it's a silent fabrication of a
plausible-looking number, in a column whose whole point is being fed downstream as a
feature.

### What to do

Compute `__length`/`__n_words` only over non-null values and leave missing rows as
`NaN` in the output, matching what `text_statistics()` already does and what
`DateTimeExpander`'s numeric extractors do for `NaT`. E.g. mask the astype(str) result
with the original null mask, or compute on `X[column].dropna()` and reindex.

### Tests

A text column with missing values; assert `__length`/`__n_words` are `NaN` (not a
number derived from the literal string `"nan"`) for those rows, and that
`text_statistics()`'s reported stats and the actual transform output agree about which
rows are missing.

**Before you start:** please comment to claim this issue, and check whether someone
already has.
